scripts/list-maintainer-workspaces.js

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env node
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs-extra';
+import { resolve } from 'path';
+import * as url from 'url';
+import * as codeowners from 'codeowners-utils';
+
+const __dirname = url.fileURLToPath(new URL('.', import.meta.url));
+
+const EXCLUDED_WORKSPACES = ['noop', 'repo-tools'];
+
+async function main() {
+  const rootPath = resolve(__dirname, '..');
+  const workspacePath = resolve(rootPath, 'workspaces');
+
+  // Get `CODEOWNERS` entries
+  const codeownersPath = resolve(rootPath, '.github', 'CODEOWNERS');
+  const codeOwnerEntries = await codeowners.loadOwners(codeownersPath);
+
+  // Get workspaces
+  const workspaces = (await fs.readdir(workspacePath, { withFileTypes: true }))
+    .filter(w => w.isDirectory() && !EXCLUDED_WORKSPACES.includes(w.name))
+    .map(w => w.name);
```

Comment:
This logic was extracted into a [dedicated function](https://github.com/backstage/community-plugins/blob/main/scripts/list-workspaces.js).

```js
import { listWorkspaces } from './list-workspaces.js';
```

```suggestion
  // Get workspaces
  const workspaces = await listWorkspaces();
```

Reply:
Nice! I've refactored to use this 👍